crates/compiler/module/src/called_via.rs

```diff
@@ -89,9 +89,22 @@ pub enum CalledVia {
     /// e.g. "$(first) $(last)" is transformed into Str.concat (Str.concat first " ") last.
     StringInterpolation,
 
-    /// This call is the result of desugaring a Record Builder field.
+    /// This call is the result of desugaring an old style Record Builder field.
     /// e.g. succeed { a <- get "a" } is transformed into (get "a") (succeed \a -> { a })
-    RecordBuilder,
+    OldRecordBuilder,
+
+    /// This call is the result of desugaring a map2-based Record Builder field. e.g.
+    /// ```roc
+    /// { Task.parallel <-
+    ///     foo: get "a",
+    ///     bar: get "b",
+    /// }
+    /// ```
+    /// is transformed into
+    /// ```roc
+    /// Task.parallel (get "a") (get "b") \foo, bar -> { foo, bar }
+    /// ```
+    NewRecordBuilder,
```

Comment:
Very minor but I'd call this just `RecordBuilder`. That way we don't have to rename it later.

Reply:
Yeah, the thought was it would make it harder for us to accidentally remove new record builder content, but we probably don't have to be that worried. All the old stuff was renamed to "old record builder", and we have tests to make sure we don't break anything. So I'll change all NewRecordBuilder names to just RecordBuilder.